src/basic/terminal-util.c

```diff
@@ -589,10 +589,21 @@ int make_console_stdio(void) {
                         return log_error_errno(r, "Failed to make /dev/null stdin/stdout/stderr: %m");
 
         } else {
+                unsigned rows, cols;
+
                 r = reset_terminal_fd(fd, true);
                 if (r < 0)
                         log_warning_errno(r, "Failed to reset terminal, ignoring: %m");
 
+                r = proc_cmdline_tty_size("/dev/console", &rows, &cols);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to get terminal size, ignoring: %m");
+                else {
+                        r = terminal_set_size_fd(fd, NULL, rows, cols);
+                        if (r < 0)
+                                log_warning_errno(r, "Failed to set terminal size, ignoring: %m");
+                }
```

Comment:
it always feels weir if we parse /proc/cmdline again during runtime from PID 1. We should do so once when initializing, and then operate on the cached values I think.

Hence the big loop that parses the kernel cmdline opts should probably parse the rows/cols values for /dev/console automatically, and then put them in the usual static `arg_xyz` variables, and then we use them everywhere.